godotproject/enemy-Ant.gd

```diff
@@ -0,0 +1,89 @@
+extends KinematicBody2D
+
+signal fireball_start
+
+const GRAVITY = 10
+const SPEED = 100
+const FLOOR = Vector2(0, -1)
+const FIREBALL = preload("res://AntAttackFire.tscn")
+
+enum STATE{
+	idle_e,
+	detected_player_e,
+	attacking_player_e
+}
+
+var GLOBAL_DEBUG = 0
+
+var velocity = Vector2(0, 0)
+var direction = -1
+
+var attackDamage = 1
+var attackState = STATE.idle_e
+var is_dead = false
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	pass
+	
+func dead():
+	is_dead = true
+	velocity = 0
+	$AnimatedSprite.play("dead")
+
+func _physics_process(delta):
+	
+	if is_dead == false:
+		if attackState == STATE.idle_e:
+			#Play Animation
+			if direction == 1:
+				$Orientation.scale.x = -1
+			else:
+				$Orientation.scale.x = 1
+				
+			$Orientation/Sprite.play("walk")
+				
+			#Movement for the ant
+			velocity.x = SPEED * direction
+			velocity.y += GRAVITY
+			velocity = move_and_slide(velocity, FLOOR)
+			
+			if is_on_wall():
+				direction *= -1
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+#func _process(delta):
+#	pass
+
+func begin_attack():
+	attackState = STATE.detected_player_e
+	$ActionDelay.start()
+
+func launch_attack():
+	if GLOBAL_DEBUG == 1:
```

Comment:
I like this, we should use it more often when printing to the console

Reply:
Useful for debugging, I don't want these print statements on main though. I will remove this for now. If we want we can have a Global_Print_Debug Flag that allows print statements in all our code though. Lets talk about this today

Reply:
sounds good

Reply:
I think we can just use the print_debug function, which shouldn't print anything to the console when we build in Release mode